src/state/api/reducer.js

```diff
@@ -0,0 +1,41 @@
+import { SET_API } from 'state/api/types';
+
+const initialState = {
+  useMocks: true,
+  domain: null,
+};
+
+const isUseMockValid = useMocks => typeof useMocks === 'boolean';
+
+const isDomainValid = (domain, useMocks) => (
+  // eslint-disable-next-line no-useless-escape
+  (useMocks && domain == null) || /^(http(s)?\:)?\/\/[a-z0-9][a-z0-9-]*(\.)?[a-z0-9-]*\.[a-z]{2,6}$/i.test(domain)
+);
+
+const isPayloadValid = (payload) => {
```

Comment:
This could be more concise this way:

```javascript
const isPayloadValid = payload => (isUseMockValid(payload.useMocks) && isDomainValid(payload.domain, payload.useMocks));
```